Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp

```diff
@@ -219,24 +219,17 @@ namespace AZ
                 Render::Bootstrap::DefaultWindowBus::Handler::BusConnect();
                 Render::Bootstrap::RequestBus::Handler::BusConnect();
 
-                // If the settings registry isn't available, something earlier in startup will report that failure.
-                if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
-                {
-                    // Automatically register the event if it's not registered, because
-                    // this system is initialized before the settings registry has loaded the event list.
-                    AZ::ComponentApplicationLifecycle::RegisterHandler(
-                        *settingsRegistry, m_componentApplicationLifecycleHandler,
-                        [this](const AZ::SettingsRegistryInterface::NotifyEventArgs&)
+                // delay one frame for Initialize which asset system is ready by then
+                AZ::TickBus::QueueFunction(
+                    [this]()
+                    {
+                        Initialize();
+                        if (m_nativeWindow)
                         {
-                            Initialize();
-                            if (m_nativeWindow)
-                            {
-                                // wait until swapchain has been created before setting fullscreen state
-                                m_nativeWindow->SetFullScreenState(r_fullscreen);
-                            }
-                        },
-                        "CriticalAssetsCompiled");
-                }
+                            // wait until swapchain has been created before setting fullscreen state
+                            m_nativeWindow->SetFullScreenState(r_fullscreen);
+                        }
+                    });
```

Comment:
Does this still work if the Cache is empty before you start the system?

Reply:
Yes it will still work. 
The  CCryEditApp::ConnectToAssetProcessor() and  CCryEditApp::CompileCriticalAssets() is called after system components activated but before handle the queued functions.